### PR TITLE
Numpy det and slogdet operator

### PR DESCRIPTION
**Pull request by ckt624**
## Description ##
Implements numpy.linalg.det and slogdet. See
https://docs.scipy.org/doc/numpy/reference/generated/numpy.linalg.det.html
https://docs.scipy.org/doc/numpy/reference/generated/numpy.linalg.slogdet.html
### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Implements numpy.linalg.det
- [x] Implements numpy.linalg.slogdet
## Comments ##
@haojin2 @reminisce

### CODE REVIEW

**Comment by haojin2** [comment]
Some tests are failing, please investigate.

File: src/operator/tensor/la_op.cc [hunk 1]
```diff
@@ -983,6 +984,7 @@ Examples::
 .add_argument("A", "NDArray-or-Symbol", "Tensor of square matrix");
 
 NNVM_REGISTER_OP(_backward_linalg_det)
+.add_alias("_backward_np_linalg_det")
```
Comment (reminisce) [inline src/operator/tensor/la_op.cc]: Adding np alias for backward op seems not necessary.

File: src/operator/tensor/la_op.cc [hunk 2]
```diff
@@ -1038,6 +1041,7 @@ Examples::
 .add_argument("A", "NDArray-or-Symbol", "Tensor of square matrix");
 
 NNVM_REGISTER_OP(_backward_linalg_slogdet)
+.add_alias("_backward_np_linalg_slogdet")
```
Comment (reminisce) [inline src/operator/tensor/la_op.cc]: Same as above.

File: src/operator/tensor/la_op.cu [hunk 3]
```diff
@@ -100,15 +100,19 @@ NNVM_REGISTER_OP(_backward_linalg_inverse)
 .set_attr<FCompute>("FCompute<gpu>", LaOpBackward<gpu, 2, 2, 2, 1, inverse_backward>);
 
 NNVM_REGISTER_OP(_linalg_det)
+.add_alias("_np__linalg_det")
 .set_attr<FCompute>("FCompute<gpu>", LaOpDetForward<gpu, 1, det>);
 
 NNVM_REGISTER_OP(_backward_linalg_det)
+.add_alias("_backward_np_linalg_det")
```
Comment (reminisce) [inline src/operator/tensor/la_op.cu]: Same as above.

File: src/operator/tensor/la_op.cu [hunk 4]
```diff
@@ -100,15 +100,19 @@ NNVM_REGISTER_OP(_backward_linalg_inverse)
...
 .set_attr<FCompute>("FCompute<gpu>", LaOpDetBackward<gpu, 1, det_backward>);
 
 NNVM_REGISTER_OP(_linalg_slogdet)
+.add_alias("_np__linalg_slogdet")
 .set_attr<FCompute>("FCompute<gpu>", LaOpDetForward<gpu, 2, slogdet>);
 
 NNVM_REGISTER_OP(_backward_linalg_slogdet)
+.add_alias("_backward_np_linalg_slogdet")
```
Comment (reminisce) [inline src/operator/tensor/la_op.cu]: Same as above.

**Comment by reminisce** [comment]
I think you will need to modify the `DetShape` function so that `np.linalg.det` returns a scalar tensor when `ndim=2`. You can just add a check of `is_np_shape` in the following and set `TShape(0, 1)` to the output shape.
https://github.com/apache/incubator-mxnet/blob/c3f5eeabf9bf9d37ac47a4f1c8cc93c292825d2a/src/operator/tensor/la_op.h#L431

File: src/operator/tensor/la_op.cc (hunk 1 above)
Comment (ckt624) [reply src/operator/tensor/la_op.cc]: Fixed. Thx.

File: src/operator/tensor/la_op.cu (hunk 3 above)
Comment (ckt624) [reply src/operator/tensor/la_op.cu]: Fixed. Thx.

File: src/operator/tensor/la_op.cu (hunk 4 above)
Comment (ckt624) [reply src/operator/tensor/la_op.cu]: Fixed. Thx.

File: python/mxnet/_numpy_op_doc.py [hunk 5]
```diff
@@ -20,6 +20,129 @@
...
+    Parameters
+    ----------
+    a : (..., M, M) ndarray
+        Input array, has to be a square 2-D array.
+
+    Returns
+    -------
+    sign : (...) ndarray
+        A number representing the sign of the determinant. For a real matrix,
+        this is 1, 0, or -1. For a complex matrix, this is a complex number
```
Comment (haojin2) [inline python/mxnet/_numpy_op_doc.py]: we do not support complex matrices yet, please fix all docs according to our actual behaviors.

File: tests/python/unittest/test_numpy_op.py [hunk 6]
```diff
@@ -213,6 +213,94 @@ def test_np_dot():
...
+                # Test imperative once again
+                mx_out = np.linalg.det(a)
+                np_out = _np.linalg.det(a.asnumpy())
+                assert_almost_equal(mx_out.asnumpy(), np_out, rtol=1e-1, atol=1e-1)
+
+                # test numeric gradient
+                a_sym = mx.sym.Variable("a").as_np_ndarray()
+                mx_sym = mx.sym.np.linalg.det(a_sym).as_nd_ndarray()
+                check_numeric_gradient(mx_sym, [a.as_nd_ndarray()],
+                    rtol=1e-1, atol=1e-1, dtype = dtype)
```
Comment (haojin2) [inline tests/python/unittest/test_numpy_op.py]: fix all spaces around `=` operators.

File: python/mxnet/_numpy_op_doc.py (hunk 5 above)
Comment (ckt624) [reply python/mxnet/_numpy_op_doc.py]: fixed. Thx.

File: tests/python/unittest/test_numpy_op.py (hunk 6 above)
Comment (ckt624) [reply tests/python/unittest/test_numpy_op.py]: fixed. Thx.

**Pull request merged**